draft-kowalik-rpp-architecture-02.adoc

```diff
@@ -392,6 +392,18 @@ Each resource type shall define operations possible on this resource type. This
 
 Operations shall be defined in IANA registry in a way that allows for automated processing of the operation definition, including constraints and references to other resource types.
 
+=== Extension mechanisms
+The RPP architecture is designed to be extensible, allowing for the addition of new resource types, data elements, and operations without breaking existing implementations. This extensibility is achieved through:
+
+* **Layered Design:** Each architectural layer (HTTP Transport, Data Representation, Resource Definition) is defined independently, allowing new features or technologies to be introduced at one layer without impacting others.
+* **IANA Registries:** Resource types, data elements, mappings, and operations are registered in IANA registries using machine-readable formats. This enables automated processing, discovery, and extension of protocol elements without requiring changes to the core specifications.
+* **Resource and Operation Extension:** New resource types and operations can be defined and registered. Existing resources can be extended with additional data elements or operations in a backward-compatible manner.

```

Comment:
* **Status codes:** New status codes can be defined and registered including their mapping on HTTP status codes.